packages/wdio-reporter/tests/reporter.listeners.test.js

```diff
@@ -540,6 +541,58 @@ describe('WDIOReporter Listeners', () => {
         })
     })
 
+    describe('handling test:retries', () => {
+
+        let runnerStartEvent
+        let suiteStartEvent
+        let testStartEvent
+        let testRetriesEvent
+        let testFailEvent
+        let testEndEvent
+
+        const emitEvents = () => {
+            reporter.emit('runner:start', runnerStartEvent)
+            reporter.emit('suite:start', suiteStartEvent)
+            reporter.emit('test:start', testStartEvent)
+            reporter.emit('test:retries', testRetriesEvent)
+            reporter.emit('test:start', testStartEvent)
+            reporter.emit('test:fail', testFailEvent)
+            reporter.emit('test:end', testEndEvent)
+        }
+
+        beforeEach(() => {
+            runnerStartEvent = {
+                cid: 'runnerid',
+                capabilities: {
+                    browserName: 'Chrome'
+                }
+            }
+            suiteStartEvent = {
+                uid: 'suiteid',
+                title: 'the software'
+            }
+            testStartEvent = {
+                uid: 'testid',
+                title: 'should do the needful'
+            }
+            testRetriesEvent = {
+                uid: 'testid'
+            }
+            testFailEvent = {
+                uid: 'testid'
+            }
+            testEndEvent = {
+                uid: 'testid'
+            }
+        })
+
+        it('should call test retry with the test stat', () => {
+            reporter.onTestEnd = jest.fn()
+            emitEvents()
+            expect(reporter.onTestEnd).toHaveBeenCalledWith(reporter.tests[testStartEvent.uid])
```

Comment:
shouldn't this check `onTestRetry`? Also could we check if `reporter.retries` was changed too?

Reply:
Updated unit tests. Added tests for `retries` count, `passes` and `failures` count before and after retry.